dotcom-rendering/src/web/components/FetchCommentCounts.importable.tsx

```diff
@@ -44,7 +44,7 @@ type RenderedCountType = {
 function extractMarkers() {
 	const markers: MarkerType[] = [];
 	document
-		.querySelectorAll('[data-name="comment-count-marker"]')
+		.querySelectorAll('[data-discussion-id]')
```

Comment:
I'm a little torn on this.

The selector is simpler, but it's slightly less explicit -- _if_ a `data-discussion-id` attribute gets added to a different kind of element at some point in the future then it would be difficult to track down why it suddenly started getting hydrated in this sort of way?

Not a super likely scenario, though, so probably not a big deal.

Reply:
I think that in general, this whole pattern is at odds with our explicit `Island`s implementation. The benefit of this proposal is that it create a tight coupling between the presence of `data-discussion-id`, IntelliSense documentation when this attribute is used in JSX and the DOM manipulation logic.

A possible improvement would be to update this whole pattern so we serialise all the required props like our usual islands, but it would be a much bigger refactor. Turning this into an `Island`, we would need to ensure that a single polling fetch is used, probably require a use of `Set`s and some shared state between islands, which is something we usually avoid.

On this specific proposal, the risk than one uses [`data-discussion-id` appears no greater](https://cs.github.com/?scopeName=All+repos&scope=&q=repo%3Aguardian%2Fdotcom-rendering+%22data-discussion-id%22) to me than the risk than one uses `data-name"comment-count-marker"` and the documentation would be a lot more verbose, but also suggest that this is a _good pattern™_, which I’m not sure it is. For example:

```ts
declare namespace JSX {
	interface IntrisicAttributes {
		/**
		 * A special name to identify markers that can modify the DOM.
		 * Acceptable values are:
		 *
		 * - 'comment-count-marker': @see {@link ../src/web/components/FetchCommentCounts.importable.tsx}
		 */
		'data-name'?: 'comment-count-marker';
	}
}
```

Reply:
okay yep this makes sense to me, thanks!